tests/integration/frameworks/test_catboost_impl.py

```diff
@@ -1,23 +1,23 @@
 import typing as t
+from typing import TYPE_CHECKING
 
 import catboost as cbt
+import numpy as np
+import psutil
 import pytest
 from catboost.core import CatBoost, CatBoostClassifier, CatBoostRegressor
 
 import bentoml.catboost
-import bentoml.models
-from bentoml._internal.models import Model
 from bentoml.exceptions import BentoMLException
 from tests.utils.frameworks.sklearn_utils import test_df
 from tests.utils.helpers import assert_have_file_extension
+from tests.utils.types import InvalidModule, Pipeline
 
-if t.TYPE_CHECKING:
-    from bentoml._internal.models import ModelStore
+if TYPE_CHECKING:
+    from bentoml._internal.models import Model, ModelStore
 
-TEST_MODEL_NAME = __name__.split(".")[-1]
 
-
-def catboost_model() -> cbt.core.CatBoostClassifier:
+def create_catboost_model() -> cbt.core.CatBoostClassifier:
```

Comment:
let's modify this to `create_catboost_model(save_params, metadata)` and replace the `pipleline` fixture?